sycl/doc/extensions/GroupMask/GroupMask.asciidoc

```diff
@@ -286,25 +285,24 @@ struct group_mask {
   void flip();
   void flip(id<1> id);
 
-  bool operator==(group_mask rhs) const;
-  bool operator!=(group_mask rhs) const;
+  bool operator==(const group_mask& rhs) const;
+  bool operator!=(const group_mask& rhs) const;
 
-  group_mask operator &=(group_mask rhs);
-  group_mask operator |=(group_mask rhs);
-  group_mask operator ^=(group_mask rhs);
-  group_mask operator <<=(size_t);
-  group_mask operator >>=(size_t rhs);
+  group_mask &operator &=(const group_mask& rhs);
+  group_mask &operator |=(const group_mask& rhs);
+  group_mask &operator ^=(const group_mask& rhs);
+  group_mask &operator <<=(size_t n);
+  group_mask &operator >>=(size_t n);
 
   group_mask operator ~() const;
-  group_mask operator <<(size_t) const;
-  group_mask operator >>(size_t) const;
+  group_mask operator <<(size_t n) const;
+  group_mask operator >>(size_t n) const;
 
+  group_mask operator &(const group_mask& rhs) const;
+  group_mask operator |(const group_mask& rhs) const;
+  group_mask operator ^(const group_mask& rhs) const;
 };
 
-group_mask operator &(const group_mask& lhs, const group_mask& rhs);
-group_mask operator |(const group_mask& lhs, const group_mask& rhs);
-group_mask operator ^(const group_mask& lhs, const group_mask& rhs);
-
```

Comment:
Do you know what the rationale was originally for defining these as free functions vs. member functions?  What was the reason for changing it?

Tagging @rolandschulz who is better versed on defining operators as hidden friend functions vs. member functions.

Reply:
I got into trouble for implementing free functions as they cannot be implemented in headers. For the host code, it fires immediately (multiple definitions) for the device code it is a problem with complex cases. Implementing device lib for them looks like overkill.

Reply:
I moved to the free friend functions